x-pack/plugin/search-business-rules/src/main/java/org/elasticsearch/xpack/searchbusinessrules/PinnedQueryBuilder.java

```diff
@@ -211,7 +218,10 @@ private PinnedQueryBuilder(QueryBuilder organicQuery, List<String> ids, List<Ite
                 }
             }
         }
-        this.ids = ids;
+
+        SetOnce<List<String>> idSupplier = new SetOnce<>();
```

Comment:
I am not sure I understand why we need to modify `PinnedQueryBuilder`? Maybe leftovers from a previous iteration?

Reply:
I had set the Supplier using a SetOnce during the re-write phase, as it seems to be a paradigm that other rewriting queries (Terms, Percolator, for example) use quite frequently. We can probably safely remove these changes based on the current working RuleBasedQuery.